### PR TITLE
#91 - feat(backend) - Add logging

### LINKED ISSUE #91: Add logging to backend
As a developer, I want to be able to log activity and/or errors coming from the Java backend using a logging framework

Size/Points: M (5)
Priority: High
Risk: Medium

Acceptance Criteria:

- Adding the dependencies to the backend to allow developers to add logging 
- Add 4 types of logs:
- INFO-level logging is implemented for key application lifecycle events, including:
Application startup and shutdown
Successful connections to external services (e.g., databases, APIs)
Successful completion of important business logic operations
- ERROR-level logging is implemented for:
All exceptions or errors that result in failed operations
Any critical failures in dependencies or services (e.g., unable to connect to the database).
- DEBUG-level logging includes additional details for:
Function entry and exit points for key service methods
Intermediate data values during complex calculations or logic (when debug mode is enabled)
- WARN-level logging is used for non-critical issues that require attention but do not stop the application from functioning (e.g., use of deprecated functionality or unexpected input values)

### PR DESCRIPTION
**Pull request by justncheng**
**Summary:**

Added logging dependencies, properties, log4j2-spring.xml file, and sample logs

**Changes Made:**

- Added Maven dependency for log4j2
- Added property file for logging to be saved in /logs folder
- Added sample logs to backend class

**Testing Instructions:**

- Select backend folder as root
- Run "mvn clean install"
- Run the main backend class, BackendApplication.java

### CODE REVIEW

**Review by im-samwong — DISMISSED** [review]
LGTM

File: apps/backend/src/main/java/com/example/backend/BackendApplication.java [hunk 1]
```diff
@@ -1,13 +1,25 @@
 package com.example.backend;
 
-import org.springframework.boot.CommandLineRunner;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+//import org.springframework.boot.CommandLineRunner;
```
Comment (XavierGuertin) [inline apps/backend/src/main/java/com/example/backend/BackendApplication.java]: Optimize Imports

File: apps/backend/src/main/java/com/example/backend/BackendApplication.java [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.context.annotation.Bean;
+//import org.springframework.context.annotation.Bean;
```
Comment (XavierGuertin) [inline apps/backend/src/main/java/com/example/backend/BackendApplication.java]: Optimize Imports

File: apps/backend/src/main/java/com/example/backend/BackendApplication.java [hunk 3]
```diff
@@ -1,13 +1,25 @@
...
 
 @SpringBootApplication
 public class BackendApplication {
+
+  private static final Logger logger = LogManager.getLogger(BackendApplication.class);
   public static void main(String[] args) {
     SpringApplication.run(BackendApplication.class, args);
+
+    logger.debug("Debug Log!");
```
Comment (XavierGuertin) [inline apps/backend/src/main/java/com/example/backend/BackendApplication.java]: Put all of this as comments instead

**Review by XavierGuertin — CHANGES_REQUESTED** [review]

File: apps/backend/src/main/java/com/example/backend/BackendApplication.java (hunk 1 above)
Comment (justncheng) [reply apps/backend/src/main/java/com/example/backend/BackendApplication.java]: Done

File: apps/backend/src/main/java/com/example/backend/BackendApplication.java (hunk 3 above)
Comment (justncheng) [reply apps/backend/src/main/java/com/example/backend/BackendApplication.java]: Done

File: apps/backend/src/main/java/com/example/backend/BackendApplication.java (hunk 2 above)
Comment (justncheng) [reply apps/backend/src/main/java/com/example/backend/BackendApplication.java]: Done

**Review by XavierGuertin — APPROVED** [review]

**Review by yasminehilout — APPROVED** [review]
Good work

**Pull request merged**